subsys/bluetooth/controller/util/mfifo.h

```diff
@@ -0,0 +1,177 @@
+
+#define MFIFO_DEFINE(name, sz, cnt) \
+		struct { \
+			u8_t const s; /* TODO: const, optimise RAM use */ \
+			u8_t const n; /* TODO: const, optimise RAM use */ \
+			u8_t f; \
+			u8_t l; \
+			u8_t m[sz * ((cnt) + 1)]; \
```

Comment:
If you are using this only to store ring buffers of pointers, then you could simplify by:
`void* m[(cnt) + 1];` and get rid of `s` and `sz` altogether.

Reply:
This would also get rid of all the multiplications in the getters/setters

Reply:
I will take relook at this implementation once I resolve all the checkpatch issues.